aten/src/ATen/native/Scalar.cpp

```diff
@@ -11,11 +11,17 @@
 #include <ATen/ops/item_native.h>
 #endif
 
+#include <torch/csrc/autograd/grad_mode.h>
+
 namespace at::native {
 
 Scalar item(const Tensor& self) {
   auto numel = self.sym_numel();
   TORCH_CHECK(numel == 1, "a Tensor with ", numel, " elements cannot be converted to Scalar");
+  if (torch::autograd::GradMode::is_enabled() && self.requires_grad()) {
```

Comment:
No need to include `torch` header, as AT has everything
```suggestion
  if (at::GradMode::is_enabled() && self.requires_grad()) {
```